test/unitary/xtypes/dynamicdata_operators.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <xtypes/xtypes.hpp>
+#include "../utils.hpp"
+
+using namespace eprosima::xtypes;
+
+#define PI 3.14159f
+
+/***********************************************
+ *         DynamicData Operators Tests         *
+ **********************************************/
+#define ASSERT_EQ_DYNAMICDATA(DYNAMICDATA, TYPE, VAL) ASSERT_EQ(DYNAMICDATA.value<TYPE>(), VAL)
+
+#define ASSERT_EQ_DYNAMICDATA_INT32(DYNAMICDATA, VAL) ASSERT_EQ_DYNAMICDATA(DYNAMICDATA, int32_t, VAL)
+
+TEST (DynamicDataOperators, increment_decrement_operators)
+{
+    DynamicData data(primitive_type<int32_t>());
+    data.value(4);
+    DynamicData& dataref = ++data;
+    ASSERT_EQ_DYNAMICDATA_INT32(dataref, 5);
```

Comment:
Why do you need the reference? I think it will be clearer just `ASSERT_EQ_DYNAMICDATA_INT32(++data, 5);`